File: backend/src/main/java/com/anilyadav/ems/service/auth/UserService.java
package com.anilyadav.ems.service.auth;

import com.anilyadav.ems.dto.request.UserRequest;
import com.anilyadav.ems.dto.response.UserResponse;

import java.util.List;

public interface UserService {

    UserResponse createUser(UserRequest userRequest);

    UserResponse getUserById(Long id);

    List<UserResponse> getAllUsers();

    UserResponse updateUser(Long id, UserRequest userRequest);

    void deleteUser(Long id);
}

File: frontend/src/services/userService.js
import {
  getAllUsersApi,
  getUserByIdApi,
  createUserApi,
  updateUserApi,
  deleteUserApi,
} from "../api/userApi";

export const getAllUsers = async () => {
  const response = await getAllUsersApi();
  return response.data;
};

export const getUserById = async (id) => {
  const response = await getUserByIdApi(id);
  return response.data;
};

export const createUser = async (user) => {
  const response = await createUserApi(user);
  return response.data;
};

export const updateUser = async (id, user) => {
  const response = await updateUserApi(id, user);
  return response.data;
};

export const deleteUser = async (id) => {
  await deleteUserApi(id);
};